zk_toolbox/crates/zk_inception/src/commands/consensus.rs

```diff
@@ -0,0 +1,365 @@
+    let mut x = vec![];
+    x.extend(k.tag);
+    x.extend(k.x);
+    ByteFmt::decode(&x)
+}
+
+fn decode_weighted_attester(
+    a: &abi::CommitteeAttester,
+) -> anyhow::Result<attester::WeightedAttester> {
+    Ok(attester::WeightedAttester {
+        weight: a.weight.into(),
+        key: decode_attester_key(&a.pub_key).context("key")?,
+    })
+}
+
+fn encode_attester_key(k: &attester::PublicKey) -> abi::Secp256K1PublicKey {
+    let b: [u8; 33] = ByteFmt::encode(k).try_into().unwrap();
+    abi::Secp256K1PublicKey {
+        tag: b[0..1].try_into().unwrap(),
+        x: b[1..33].try_into().unwrap(),
+    }
+}
+
+fn encode_validator_key(k: &validator::PublicKey) -> abi::Bls12381PublicKey {
+    let b: [u8; 96] = ByteFmt::encode(k).try_into().unwrap();
+    abi::Bls12381PublicKey {
+        a: b[0..32].try_into().unwrap(),
+        b: b[32..64].try_into().unwrap(),
+        c: b[64..96].try_into().unwrap(),
+    }
+}
+
+fn encode_validator_pop(pop: &validator::ProofOfPossession) -> abi::Bls12381Signature {
+    let b: [u8; 48] = ByteFmt::encode(pop).try_into().unwrap();
+    abi::Bls12381Signature {
+        a: b[0..32].try_into().unwrap(),
+        b: b[32..48].try_into().unwrap(),
+    }
+}
+
+#[derive(clap::Subcommand, Debug)]
+pub enum Command {
+    /// Sets the attester committee in the consensus registry contract to
+    /// `consensus.genesis_spec.attesters` in general.yaml.
+    SetAttesterCommittee,
+    /// Fetches the attester committee from the consensus registry contract.
+    GetAttesterCommittee,
+}
+
+/// Collection of sent transactions.
+#[derive(Default)]
+pub struct TxSet(Vec<(H256, &'static str)>);
+
+impl TxSet {
+    /// Sends a transactions and stores the transaction hash.
+    pub async fn send<M: 'static + Middleware, B: Borrow<M>, D: Detokenize>(
+        &mut self,
+        name: &'static str,
+        call: FunctionCall<B, M, D>,
+    ) -> anyhow::Result<()> {
+        let h = call.send().await.context(name)?.tx_hash();
+        self.0.push((h, name));
+        Ok(())
+    }
+
+    /// Waits for all stored transactions to complete.
+    pub async fn wait<P: JsonRpcClient>(self, provider: &Provider<P>) -> anyhow::Result<()> {
+        for (h, name) in self.0 {
+            async {
+                let status = PendingTransaction::new(h, provider)
+                    .await
+                    .context("await")?
+                    .context(messages::MSG_RECEIPT_MISSING)?
+                    .status
+                    .context(messages::MSG_STATUS_MISSING)?;
+                anyhow::ensure!(status == 1.into(), messages::MSG_TRANSACTION_FAILED);
+                Ok(())
+            }
+            .await
+            .context(name)?;
+        }
+        Ok(())
+    }
+}
+
+fn print_attesters(committee: &attester::Committee) {
+    logger::success(
+        committee
+            .iter()
+            .map(|a| format!("{a:?}"))
+            .collect::<Vec<_>>()
+            .join("\n"),
+    );
```

Comment:
nit: why not implementing Display for committee?

Reply:
It is defined in different repo.